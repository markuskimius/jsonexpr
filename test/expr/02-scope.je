i = 1010;

FUNCTION("",
    j = 2020;

    FUNCTION("",
        UPSCOPE.i = 2010;

        PRINT("Level 3", i++, j++);
    )();

    PRINT("Level 2", i++, j++);
)();

PRINT("Level 1", i);
PRINT(j);
