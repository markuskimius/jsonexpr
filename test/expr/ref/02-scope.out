Level 3 2010 2020
Level 2 2011 2021
Level 1 1010
JE_RuntimeError at line 16 col 7 near `j`: No such variable named "j" in this scope
null
